modules/dxp/apps/segments/segments-experiment-web/src/main/resources/META-INF/resources/js/components/ClickGoalPicker/ClickGoalPicker.es.js

```diff
@@ -345,14 +355,14 @@ function Overlay({allowEdit, root, targetableElements}) {
 					if (allowEdit === true) {
 						return true;
 					}
-					if ('#' + element.id === selectedTarget) {
+					if (element.id === selectedTarget) {
```

Comment:
This whole thing is probably ripe for simplication now:

```js
.filter((element) => (
    allowEdit === true || element.id === selectedTarget
))
```

This would be better too if `allowEdit` is reliably truthy/falsey (I haven't looked at the type because I am lazy):

```js
.filter((element) => (
    allowEdit || element.id === selectedTarget
))
```

Reply:
That's right, I forgot to review the conditionals, thanks!